src/StateSpace.hpp

```diff
@@ -28,6 +28,58 @@ class StateSpace {
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control. A previous
+     * stepsize of 0 will indicate that the this is the first iteration of the tree and will
+     * automatically set the stepsize to defaultStepSize.
+     *
+     * @param source The node in the tree to extend from
+     * @param target The point in the space to extend to
+     * @param prevStepSize The stepsize taken to reach source
+     * @param ascGrowthRate How aggressively the RRT grows or shrinks
+     * @param defaultStepSize The default stepsize of the RRT\
+     *
+     * @return A state in the direction of @target from @source.state()
+     */
+    virtual T intermediateState(const T &source, const T &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const = 0;
+
+    /**
+     * Sets the minimum possible stepsize the tree can extend when running adaptive stepsize control.
+     *
+     * @param minStepSize The minimum stepsize.
+     */
+    virtual void setMinStepSize(float minStepSize) = 0;
+
+    /**
+     * @return The minimum stepsize the tree can extend when running adaptive stepsize control.
+     */
+    virtual float minStepSize() const = 0;
+
+    /**
+     * Sets the maximum possible stepsize the tree can extend when running adaptive stepsize control.
+     *
+     * @param maxStepSize The maximum stepsize.
+     */
+    virtual void setMaxStepSize(float maxStepSize) = 0;
+
+    /**
+     * @return The maximum stepsize the tree can extend when running adaptive stepsize control.
+     */
+    virtual float maxStepSize() const = 0;
+
+    /**
+     * When running adaptive stepsize control, determines the maximum distance an obstacle
+     * can be from a source node for the stepSize to grow instead of shrink.
+     *
+     * @param maxDist How close the obstacles can be to the path to cause a shrink in stepsize
+     */
+    virtual void setDistScale(float distScale) = 0;
```

Comment:
The comment here says that `distScale` is the distance that determines whether to grow or shrink the step size, while the `intermediateState` method is actually multiplying it by the previous step size.